tests/test_gui_editor_browser.py

```diff
@@ -0,0 +1,415 @@
+#!/usr/bin/env python3
```

Comment:
This doesn't seem to actually be testing the browser. In all 3 integration test suites, we're testing the backend logic (validating responses etc), but not really validating the validity of the HTML. We need to exercise it in an actual browser. Use a framework like Selenium to perform actual browser testing where we render the GUI and perform actions like a user would